lib/ff_config.c

```diff
@@ -398,11 +546,43 @@ ff_check_config(struct ff_config *cfg)
 
     int i;
     for (i = 0; i < cfg->dpdk.nb_ports; i++) {
-        struct ff_port_cfg *pc = &cfg->dpdk.port_cfgs[i];
+        uint16_t portid = cfg->dpdk.portid_list[i];
+        struct ff_port_cfg *pc = &cfg->dpdk.port_cfgs[portid];
         CHECK_VALID(addr);
         CHECK_VALID(netmask);
         CHECK_VALID(broadcast);
         CHECK_VALID(gateway);
+        // check if the lcores in lcore_list are enabled.
+        int k;
+        for (k = 0; k < pc->nb_lcores; k++) {
+            uint16_t lcore_id = pc->lcore_list[k];
+            if (uint16_binary_search(cfg->dpdk.proc_lcore, 0,
+                                     cfg->dpdk.nb_ports-1, lcore_id) < 0) {
```

Comment:
`cfg->dpdk.nb_ports-1` =====> `cfg->dpdk.nb_procs-1`

Reply:
oh, it's my false, I bank on the auto-completion of IDE too much.